src/fft.rg

```diff
@@ -172,105 +241,279 @@ function fft.generate_fft_interface(itype, dtype)
+      regentlib.assert(address_space == i, "make_plan_gpu must be executed on a processor in the same address space")
+
+      -- Get input and output bases
+      var input_base = get_base_in(rect_in_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0]).base
+      var output_base = get_base_out(rect_out_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0]).base
+      var lo = input.ispace.bounds.lo:to_point()
+      var hi = input.ispace.bounds.hi:to_point()
+      var n : int[dim] -- n is an array of size dim with the size of each dimension in the entries
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+
+      -- Create plans
+      if float_to_complex32_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_R2C, 1))
+      elseif complex32_to_complex32_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_C2C, 1))
+      elseif double_to_complex64_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_D2Z, 1))
+      elseif complex64_to_complex64_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_Z2Z, 1))
+      end
     end
   end
 
-  -- Important: overwrites input/output!
-  __demand(__inline)
-  task iface.make_plan(input : region(ispace(itype), dtype),
-                       output : region(ispace(itype), dtype),
+  --- Creates the FFT plan. This is the __inline version of the task should the user wish to use that. Entry point into GPU functionality as well.
+  -- @param input Input region.
+  -- @param output Output region.
+  -- @param plan Plan region.
+  -- @note Calls `make_plan_gpu` if necessary.
+   __demand(__inline)
+  task iface.make_plan(input : region(ispace(itype), dtype_in),
+                       output : region(ispace(itype), dtype_out),
                        plan : region(ispace(int1d), iface.plan))
   where reads writes(input, output, plan) do
+
     var p = iface.get_plan(plan, false)
+    var address_space = c.legion_processor_address_space(get_executing_processor(__runtime()))
+    regentlib.assert(input.ispace.bounds == output.ispace.bounds, "input and output regions must be identical in size")
+
+    var input_base = get_base_in(rect_in_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0]).base
+    var output_base = get_base_out(rect_out_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0]).base
+    var lo = input.ispace.bounds.lo:to_point()
+    var hi = input.ispace.bounds.hi:to_point()
 
-    var address_space = c.legion_processor_address_space(
-      get_executing_processor(__runtime()))
+    var n : int[dim]
+    ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
```

Comment:
Same about stray comments. Either we need this or we don't, and if the latter, just get rid of it.

Reply:
Same as above. removing the semicolon but keeping this line